### PR TITLE
replace music backend enumeration with generic string

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

**Comment by rfomin** [comment]
I don't understand, what was wrong?

**Comment by fabiangreffrath** [comment]
There is no "Native" backend on Linux and "SDL2" is wrong anyway, FluidSynth started as `1` although it should be `0`, etc.

File: src/m_misc.c [hunk 1]
```diff
@@ -456,22 +456,22 @@ default_t defaults[] = {
...
-    "0 for SDL2 (default), "
+    xstr(__COUNTER__) " for Native (default), "
 #endif
 #if defined(HAVE_FLUIDSYNTH)
-    "1 for FluidSynth, 2 for OPL Emulation"
-#else
-    "1 for OPL Emulation"
+    xstr(__COUNTER__) " for FluidSynth, "
 #endif
+    xstr(__COUNTER__) " for OPL Emulation"
```
[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply src/m_misc.c:474]: All relevant compilers support it.

[the remark this answers is not in the document]
Comment (rfomin) [reply src/m_misc.c:474]: I suggest replacing this help string with something generic like "current music backend". I think this macro trick is too complicated.

**Pull request merged**